File: proto-csharp/ProtoMessage/ProtoMessage/ApeMessage/Reflect/LongFieldDescriptor.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using Google.ProtocolBuffers;
using com.spaceape.protobuf;

namespace com.spaceape.protobuf.reflect
{
  public class LongFieldDescriptor: FieldDescriptor
  {
    public LongFieldDescriptor(int fieldNumber, String fieldName, Type clazz) : base(fieldNumber, fieldName, clazz)
    {
    }

    override public Object readField(CodedInputStream input, CodeGenContext context)
    {
      return input.ReadInt64();
    }

    override public void writeField(CodedOutputStream output, CodeGenContext context, Object obj)
    {
      output.WriteInt64(fieldNumber, (long)obj);
    }
  }
}

File: proto-java/src/main/java/com/spaceape/protobuf/reflect/LongFieldDescriptor.java
package com.spaceape.protobuf.reflect;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.spaceape.protobuf.CodeGenContext;

import java.io.IOException;

public class LongFieldDescriptor extends FieldDescriptor {
    public LongFieldDescriptor(int fieldNumber, String fieldName, Class clazz){
        super(fieldNumber, fieldName, clazz, long.class);
    }
    public Long readField(CodedInputStream input, CodeGenContext context) throws IOException {
        return input.readInt64();
    }
    public void writeField(CodedOutputStream output, CodeGenContext context, Object object) throws IOException{
        output.writeInt64(fieldNumber, (long)object);
    }
}
